### DISCUSSION TITLE
Are the static sets thread safe?

### DISCUSSION THREAD

**Discussion by HugoPhibbs** [NVIDIA/cuCollections#593] [Q&A] [answered]
Basically the title. 

My idea is to run a multi-threaded CPU program that uses a list of static_sets. There may be conflicts across threads around reading and writing to these static sets. Essentially, you may get two threads attempting to write to the same static set at a given point in time.

If you need more info, please let me know.

Is this safe to do?

**Comment by sleeepyjack** [comment, marked as answer]
Thanks for reaching out to us!

Using the host bulk APIs is thread-safe as long as different threads perform the same kind of operation, e.g., all `insert` or all do a `find`. Mixing mutating ops (`insert`) with any lookup operation is not supported as of now.
